src/order/directed.lean

```diff
@@ -161,6 +162,72 @@ by assumption
 instance order_dual.is_directed_le [has_le α] [is_directed α (≥)] : is_directed αᵒᵈ (≤) :=
 by assumption
 
+section reflexive
+
+lemma directed_on.insert (h : reflexive r) (a : α) (s : set α) (hd : directed_on r s)
+(ha : ∀ b ∈ s, ∃ c ∈ s, a ≼ c ∧ b ≼ c) : directed_on r (insert a s) :=
+begin
+  rw directed_on,
+  intros,
+  rw [set.mem_insert_iff] at H,
+  rw [set.mem_insert_iff] at H_1,
```

Comment:
Please use explicit names in the `intros` to avoid having `H` and `H_1`!

Reply:
Done.